src/activities/activity.clj

```diff
@@ -17,39 +17,51 @@
   (s/keys :req [:crux.db/id ::title ::date-time ::creator ::duration ::capacity]
           :opts [::description ::participants]))
 
-;; TODO fix namespaces?
-(defn req->new-activity
-  "Takes a request with form input for a new activity and returns a properly
-  namespaced activity map."
-  [{{:keys [title description datetime duration capacity]} :params
-    {:keys [identity]} :session}]
-  (let [new-uuid (UUID/randomUUID)
-        inst     (utils/datetime->inst datetime)
+(defn new-activity
+  "Takes a map of activity attributes and returns a properly namespaced and
+  conformed activity map with some defaults filled in."
+  [{:keys [uuid creator title description datetime duration capacity]}]
+  (let [inst     (utils/datetime->inst datetime) ;There is a NullPointerException coming from this line
         duration (Long/parseLong duration)
         capacity (Long/parseLong capacity)]
-    {:crux.db/id            new-uuid
-     :activity/creator      identity
-     :activity/title        title
-     :activity/description  description
-     :activity/date-time    inst
-     :activity/duration     duration
-     :activity/capacity     capacity
-     :activity/participants #{}}))
+    (s/conform :activities/activity {:crux.db/id    uuid
+                                     ::creator      creator
+                                     ::title        title
+                                     ::description  description
+                                     ::date-time    inst
+                                     ::duration     duration
+                                     ::capacity     capacity
+                                     ::participants #{}})))
 
-(defn req->activity
-  "Takes a request containing an activity id in the path and returns the
-  associated entity in the database."
-  [req]
-  (let [db       (crux/db (:crux req))
-        id       (get-in req [:path-params :id])
-        uuid     (UUID/fromString id)
-        activity (crux/entity db uuid)]
-    activity))                          ;TODO validate with spec
+(defn find-activities
```

Comment:
i would name it `list-issues`

Reply:
I believe you meant `list-activities`, right? If so, it's fixed.